### ISSUE TITLE
Prioritized DQN experiment nonfunctional

### ISSUE THREAD

**Issue by CasBex** [JuliaReinforcementLearning/ReinforcementLearning.jl#971]
The Prioritzed DQN experiment is not working anymore. I get the error underneath when running. The only installed package here is ReinforcementLearningExperiments.

I'll have a look into it and send a PR if I find the issue.
```
ERROR: BoundsError: attempt to access 2×32 Matrix{Float32} at index [CartesianIndex{2}[CartesianIndex(2, 1), CartesianIndex(0, 2), CartesianIndex(2, 3), CartesianIndex(1, 4), CartesianIndex(2, 5), CartesianIndex(1, 6), CartesianIndex(2, 7), CartesianIndex(1, 8), CartesianIndex(2, 9), CartesianIndex(2, 10)  …  CartesianIndex(2, 23), CartesianIndex(2, 24), CartesianIndex(2, 25), CartesianIndex(2, 26), CartesianIndex(2, 27), CartesianIndex(2, 28), CartesianIndex(2, 29), CartesianIndex(2, 30), CartesianIndex(2, 31), CartesianIndex(2, 32)]]
Stacktrace:
  [1] throw_boundserror(A::Matrix{Float32}, I::Tuple{Vector{CartesianIndex{2}}})
    @ Base ./abstractarray.jl:744
  [2] checkbounds
    @ ./abstractarray.jl:709 [inlined]
  [3] _getindex
    @ ./multidimensional.jl:860 [inlined]
  [4] getindex(A::Matrix{Float32}, I::Vector{CartesianIndex{2}})
    @ Base ./abstractarray.jl:1294
  [5] adjoint
    @ ~/.julia/packages/Zygote/4SSHS/src/lib/array.jl:44 [inlined]
  [6] _pullback
    @ ~/.julia/packages/ZygoteRules/OgCVT/src/adjoint.jl:66 [inlined]
  [7] _pullback
    @ ~/.julia/packages/ReinforcementLearningZoo/BI1Or/src/algorithms/dqns/prioritized_dqn.jl:60 [inlined]  [8] _pullback(::Zygote.Context{true}, ::ReinforcementLearningZoo.var"#34#37"{PrioritizedDQNLearner{Approximator{TwinNetwork{Flux.Chain{Tuple{Flux.Dense{typeof(NNlib.relu), Matrix{Float32}, Vector{Float32}}, Flux.Dense{typeof(NNlib.relu), Matrix{Float32}, Vector{Float32}}, Flux.Dense{typeof(identity), Matrix{Float32}, Vector{Float32}}}}, Flux.Chain{Tuple{Flux.Dense{typeof(NNlib.relu), Matrix{Float32}, Vector{Float32}}, Flux.Dense{typeof(NNlib.relu), Matrix{Float32}, Vector{Float32}}, Flux.Dense{typeof(identity), Matrix{Float32}, Vector{Float32}}}}}, Flux.Optimise.Adam}, StableRNGs.LehmerRNG}, Vector{Float32}, Vector{Float64}, Vector{Float64}, Int64, Matrix{Float32}, ReinforcementLearningExperiments.var"#4#5", Float32, Flux.Chain{Tuple{Flux.Dense{typeof(NNlib.relu), Matrix{Float32}, Vector{Float32}}, Flux.Dense{typeof(NNlib.relu), Matrix{Float32}, Vector{Float32}}, Flux.Dense{typeof(identity), Matrix{Float32}, Vector{Float32}}}}})
    @ Zygote ~/.julia/packages/Zygote/4SSHS/src/compiler/interface2.jl:0
  [9] pullback(f::Function, ps::Zygote.Params{Zygote.Buffer{Any, Vector{Any}}})
    @ Zygote ~/.julia/packages/Zygote/4SSHS/src/compiler/interface.jl:414
 [10] gradient(f::Function, args::Zygote.Params{Zygote.Buffer{Any, Vector{Any}}})
    @ Zygote ~/.julia/packages/Zygote/4SSHS/src/compiler/interface.jl:96
 [11] optimise!(learner::PrioritizedDQNLearner{Approximator{TwinNetwork{Flux.Chain{Tuple{Flux.Dense{typeof(NNlib.relu), Matrix{Float32}, Vector{Float32}}, Flux.Dense{typeof(NNlib.relu), Matrix{Float32}, Vector{Float32}}, Flux.Dense{typeof(identity), Matrix{Float32}, Vector{Float32}}}}, Flux.Chain{Tuple{Flux.Dense{typeof(NNlib.relu), Matrix{Float32}, Vector{Float32}}, Flux.Dense{typeof(NNlib.relu), Matrix{Float32}, Vector{Float32}}, Flux.Dense{typeof(identity), Matrix{Float32}, Vector{Float32}}}}}, Flux.Optimise.Adam}, StableRNGs.LehmerRNG}, batch::NamedTuple{(:key, :priority, :state, :next_state, :action, :reward, :terminal), Tuple{Vector{Int64}, Vector{Float64}, Matrix{Float32}, Matrix{Float32}, Vector{Int64}, Vector{Float32}, Vector{Bool}}})
    @ ReinforcementLearningZoo ~/.julia/packages/ReinforcementLearningZoo/BI1Or/src/algorithms/dqns/prioritized_dqn.jl:59
 [12] optimise!(learner::PrioritizedDQNLearner{Approximator{TwinNetwork{Flux.Chain{Tuple{Flux.Dense{typeof(NNlib.relu), Matrix{Float32}, Vector{Float32}}, Flux.Dense{typeof(NNlib.relu), Matrix{Float32}, Vector{Float32}}, Flux.Dense{typeof(identity), Matrix{Float32}, Vector{Float32}}}}, Flux.Chain{Tuple{Flux.Dense{typeof(NNlib.relu), Matrix{Float32}, Vector{Float32}}, Flux.Dense{typeof(NNlib.relu), Matrix{Float32}, Vector{Float32}}, Flux.Dense{typeof(identity), Matrix{Float32}, Vector{Float32}}}}}, Flux.Optimise.Adam}, StableRNGs.LehmerRNG}, #unused#::PostActStage, trajectory::Trajectory{EpisodesBuffer{(:key, :priority, :state, :next_state, :action, :reward, :terminal), Tuple{Int64, Float32, SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}, SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}, Trace{CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, SubArray{Int64, 0, CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, SubArray{Float32, 0, CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, SubArray{Bool, 0, CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, Tuple{Int64}, true}}}, CircularPrioritizedTraces{CircularArraySARTSTraces{Tuple{MultiplexTraces{(:state, :next_state), Trace{CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}}, SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, SubArray{Int64, 0, CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, SubArray{Float32, 0, CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, SubArray{Bool, 0, CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, Tuple{Int64}, true}}}, 5, Tuple{SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}, SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}, Trace{CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, SubArray{Int64, 0, CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, SubArray{Float32, 0, CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, SubArray{Bool, 0, CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, Tuple{Int64}, true}}}}, (:key, :priority, :state, :next_state, :action, :reward, :terminal), Tuple{Int64, Float32, SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}, SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}, Trace{CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, SubArray{Int64, 0, CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, SubArray{Float32, 0, CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, SubArray{Bool, 0, CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, Tuple{Int64}, true}}}}, DataStructures.CircularBuffer{Int64}, DataStructures.CircularBuffer{Bool}}, NStepBatchSampler{(:state, :next_state, :action, :reward, :terminal)}, InsertSampleRatioController, typeof(identity)})
    @ ReinforcementLearningZoo ~/.julia/packages/ReinforcementLearningZoo/BI1Or/src/algorithms/dqns/prioritized_dqn.jl:76
 [13] optimise!
    @ ~/.julia/packages/ReinforcementLearningCore/6OIvm/src/policies/q_based_policy.jl:46 [inlined]
 [14] optimise!
    @ ~/.julia/packages/ReinforcementLearningCore/6OIvm/src/policies/agent/agent_base.jl:33 [inlined]
 [15] optimise!
    @ ~/.julia/packages/ReinforcementLearningCore/6OIvm/src/policies/agent/agent_base.jl:32 [inlined]
 [16] macro expansion
    @ ~/.julia/packages/TimerOutputs/RsWnF/src/TimerOutput.jl:253 [inlined]
 [17] _run(policy::Agent{QBasedPolicy{PrioritizedDQNLearner{Approximator{TwinNetwork{Flux.Chain{Tuple{Flux.Dense{typeof(NNlib.relu), Matrix{Float32}, Vector{Float32}}, Flux.Dense{typeof(NNlib.relu), Matrix{Float32}, Vector{Float32}}, Flux.Dense{typeof(identity), Matrix{Float32}, Vector{Float32}}}}, Flux.Chain{Tuple{Flux.Dense{typeof(NNlib.relu), Matrix{Float32}, Vector{Float32}}, Flux.Dense{typeof(NNlib.relu), Matrix{Float32}, Vector{Float32}}, Flux.Dense{typeof(identity), Matrix{Float32}, Vector{Float32}}}}}, Flux.Optimise.Adam}, StableRNGs.LehmerRNG}, EpsilonGreedyExplorer{:exp, false, StableRNGs.LehmerRNG}}, Trajectory{EpisodesBuffer{(:key, :priority, :state, :next_state, :action, :reward, :terminal), Tuple{Int64, Float32, SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}, SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}, Trace{CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, SubArray{Int64, 0, CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, SubArray{Float32, 0, CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, SubArray{Bool, 0, CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, Tuple{Int64}, true}}}, CircularPrioritizedTraces{CircularArraySARTSTraces{Tuple{MultiplexTraces{(:state, :next_state), Trace{CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}}, SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, SubArray{Int64, 0, CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, SubArray{Float32, 0, CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, SubArray{Bool, 0, CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, Tuple{Int64}, true}}}, 5, Tuple{SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}, SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}, Trace{CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, SubArray{Int64, 0, CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, SubArray{Float32, 0, CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, SubArray{Bool, 0, CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, Tuple{Int64}, true}}}}, (:key, :priority, :state, :next_state, :action, :reward, :terminal), Tuple{Int64, Float32, SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}, SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}, Trace{CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, SubArray{Int64, 0, CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, SubArray{Float32, 0, CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, SubArray{Bool, 0, CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, Tuple{Int64}, true}}}}, DataStructures.CircularBuffer{Int64}, DataStructures.CircularBuffer{Bool}}, NStepBatchSampler{(:state, :next_state, :action, :reward, :terminal)}, InsertSampleRatioController, typeof(identity)}}, env::ReinforcementLearningEnvironments.CartPoleEnv{Float32, Int64}, stop_condition::StopAfterStep{ProgressMeter.Progress}, hook::TotalRewardPerEpisode{Val{true}, Float64}, reset_condition::ResetAtTerminal)
    @ ReinforcementLearningCore ~/.julia/packages/ReinforcementLearningCore/6OIvm/src/core/run.jl:106
 [18] run
    @ ~/.julia/packages/ReinforcementLearningCore/6OIvm/src/core/run.jl:75 [inlined]
 [19] run(policy::Agent{QBasedPolicy{PrioritizedDQNLearner{Approximator{TwinNetwork{Flux.Chain{Tuple{Flux.Dense{typeof(NNlib.relu), Matrix{Float32}, Vector{Float32}}, Flux.Dense{typeof(NNlib.relu), Matrix{Float32}, Vector{Float32}}, Flux.Dense{typeof(identity), Matrix{Float32}, Vector{Float32}}}}, Flux.Chain{Tuple{Flux.Dense{typeof(NNlib.relu), Matrix{Float32}, Vector{Float32}}, Flux.Dense{typeof(NNlib.relu), Matrix{Float32}, Vector{Float32}}, Flux.Dense{typeof(identity), Matrix{Float32}, Vector{Float32}}}}}, Flux.Optimise.Adam}, StableRNGs.LehmerRNG}, EpsilonGreedyExplorer{:exp, false, StableRNGs.LehmerRNG}}, Trajectory{EpisodesBuffer{(:key, :priority, :state, :next_state, :action, :reward, :terminal), Tuple{Int64, Float32, SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}, SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}, Trace{CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, SubArray{Int64, 0, CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, SubArray{Float32, 0, CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, SubArray{Bool, 0, CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, Tuple{Int64}, true}}}, CircularPrioritizedTraces{CircularArraySARTSTraces{Tuple{MultiplexTraces{(:state, :next_state), Trace{CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}}, SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, SubArray{Int64, 0, CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, SubArray{Float32, 0, CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, SubArray{Bool, 0, CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, Tuple{Int64}, true}}}, 5, Tuple{SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}, SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}, Trace{CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, SubArray{Int64, 0, CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, SubArray{Float32, 0, CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, SubArray{Bool, 0, CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, Tuple{Int64}, true}}}}, (:key, :priority, :state, :next_state, :action, :reward, :terminal), Tuple{Int64, Float32, SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}, SubArray{Float32, 1, CircularArrayBuffers.CircularArrayBuffer{Float32, 2, Matrix{Float32}}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}, Trace{CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, SubArray{Int64, 0, CircularArrayBuffers.CircularVectorBuffer{Int64, Vector{Int64}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, SubArray{Float32, 0, CircularArrayBuffers.CircularVectorBuffer{Float32, Vector{Float32}}, Tuple{Int64}, true}}, Trace{CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, SubArray{Bool, 0, CircularArrayBuffers.CircularVectorBuffer{Bool, Vector{Bool}}, Tuple{Int64}, true}}}}, DataStructures.CircularBuffer{Int64}, DataStructures.CircularBuffer{Bool}}, NStepBatchSampler{(:state, :next_state, :action, :reward, :terminal)}, InsertSampleRatioController, typeof(identity)}}, env::ReinforcementLearningEnvironments.CartPoleEnv{Float32, Int64}, stop_condition::StopAfterStep{ProgressMeter.Progress}, hook::TotalRewardPerEpisode{Val{true}, Float64})
    @ ReinforcementLearningCore ~/.julia/packages/ReinforcementLearningCore/6OIvm/src/core/run.jl:74
 [20] run(ex::Experiment{:JuliaRL_PrioritizedDQN_CartPole})
    @ ReinforcementLearningCore ~/.julia/packages/ReinforcementLearningCore/6OIvm/src/core/run.jl:63
 [21] top-level scope
    @ REPL[4]:1
 [22] top-level scope
    @ ~/.julia/packages/CUDA/35NC6/src/initialization.jl:190
```

**Comment by CasBex** [comment]
The 0 is added in the following function call in RLCore `agent_base.jl`
```julia
function Base.push!(agent::Agent, ::PreEpisodeStage, env::AbstractEnv)
    push!(agent.trajectory, (state = state(env),))
end
```
This also happens in the DQN case, but there the a=0 samples do not show up in the batches (even if I increase the batch size significantly). So the most likely culprit seems to be the CircularPrioritizedTraces function which allows a=0 to be taken up in the batches. (this is seems to be the main difference between regular and prioritised DQN)

**Comment by CasBex** [comment]
I have tracked the bug down to the `Random.rand(::AbstractRNG, ::SumTree, ::Int)` method in [ReinforcementLearningTrajectories.jl](https://github.com/JuliaReinforcementLearning/ReinforcementLearningTrajectories.jl/blob/main/src/common/sum_tree.jl) . This function sometimes gives samples with priority zero which should not be possible. I will take it up further in that repository.

**Comment by CasBex** [comment]
https://github.com/JuliaReinforcementLearning/ReinforcementLearningTrajectories.jl/pull/60 should fix this issue

**Comment by HenriDeh** [comment]
> This function sometimes gives samples with priority zero which should not be possible. I will take it up further in that repository.

I'll jump in late in the discussion just to ensure you understand why there are action 0's in the trajectory. This feature was recently added to RLTrajectories as a padding workaround because there were incorrect transitions sampled in the previous versions. Normally, they cannot be sampled so the 0s are just placeholders. 

I also stumbled on that problem. However, you seem to have tracked down the issue to something different than I did in https://github.com/JuliaReinforcementLearning/ReinforcementLearningTrajectories.jl/pull/57.
